Add transitive dependencies to generated Bazel BUILD files

sburton84 opened the pull request:
Changelog: Bugfix: Add transitive dependencies to generated Bazel BUILD files.
Docs: omit

Fixes #10685

Previously the Bazel generator wasn't adding transitive dependencies to the `deps` field of the BUILD file for each dependency. This results in Bazel's build graph being incorrect and causing issues with things like linking order. This change fixes #10685 by inserting these dependencies.

sburton84 committed:
Add transitive dependencies to cc_library declaration (#10685)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=10686) <br/>All committers have signed the CLA.

memsharded reviewed (approved):
Without understanding bazel, this looks good, and thanks for also providing some test.

I think this can be merged, and next necessary thing from our side is to implement a full functional test with transitive real compiled libraries to ensure everything works.

memsharded commented:
Thanks @sburton84 for contributing.

We are still supporting Py27 (until 2.0), so https://ci.conan.io/blue/organizations/jenkins/ConanTestSuite/detail/PR-10686/1/pipeline build is broken. I think it will be ok if you use ``.format()`` instead of f-strings.

sburton84 commented:
Oops, the f-strings were from a previous experiment with the test and don't even need formatting any more, so I've just removed the `f`. I also rebased into a single commit and added the issue # to the commit message.

sburton84 commented:
Just wondering if I should change the for loop to a list comprehension? Or is it okay like it is?

memsharded commented:
Hi @sburton84 

No, thanks, it is fine, scheduled for 1.47. It was just mostly waiting for other open PRs and issues related Bazel (as it is pretty new generator, it is expected). But lets merge it now, thanks for your contribution!
